DataFormats/TrackingRecHitSoA/interface/TrackingRecHitsDevice.h

```diff
@@ -47,10 +47,10 @@ namespace reco {
       auto hitsView = this->template view<TrackingRecHitSoA>();
       auto modsView = this->template view<HitModuleSoA>();
 
-      auto nModules = clusters.view().metadata().size();
+      // auto nModules = clusters.view().metadata().size();
 
-      auto clusters_m = cms::alpakatools::make_device_view(queue, clusters.view().clusModuleStart(), nModules);
-      auto hits_m = cms::alpakatools::make_device_view(queue, modsView.moduleStart(), nModules);
+      auto clusters_m = cms::alpakatools::make_device_view(queue, clusters.view().clusModuleStart());
+      auto hits_m = cms::alpakatools::make_device_view(queue, modsView.moduleStart());
```

Comment:
What is the interplay of using `make_device_view` with a `span` ?
Looking at the code in `HeterogeneousCore/AlpakaInterface/interface/memory.h` I think it will create a buffer with a single element that contains the `span` - not a buffer with `span.size()` elements.

Or did I miss something ?

Reply:
Ah, sorry, I forgot that you added the overload in #48314 👍🏻